### ISSUE TITLE
Repeatable custom field inside repeatable group don't save properly

### ISSUE THREAD

**Issue by bomsn** [CMB2/CMB2#1035]
I created custom field based on the address field
### Expected Behavior:
- The data need to be saved properly, each field created need to be saved.

### Actual Behavior:
- When adding repeatable custom field, say 3 for example, after save I only find 2.
- If there are for example 3 fields, and I added 4 other fields, after save, I only find 4 where the total should be 7
- Sometimes it will save all fields except the last one
- Saving doesn't work properly in general.

### Steps to reproduce (I have confirmed I can reproduce this issue on the trunk branch):

1. Create New Post
2. Scroll to Book details metabox, 
3. Create 3 fields (or any other number), give them random names
4. Update post (you'll see that only first 2 fields were saved)
5. Now add another 3 fields and update (you'll see that only the first one of them was saved)
6. If you add another group it will act the same way

I suspect it's javascript bug.

### CMB2 Field Registration Code:

https://gist.github.com/bomsn/36174de42bfbaa74abcd5bce1bfc432d

**Comment by bomsn** [comment]
When I removed label line from all custom fields it did work 
`<p><label for="<?php echo $field_type_object->_id( '_book_xxx' ); ?>'"></label></p>`

But this time there is another issue, it will save all field in first group correctly, but will remove last field from second group, and create a new empty group with empty field, I tried removing all HTML, but this time it will create a new group for each field when second group is saved.

I also noticed if you click the "remove field" button on the an empty field (the only field in a group) it will create an empty field in the previous group.

**Comment by bomsn** [comment]
I was able to solve this by doing some changes in `cmb.elReplacements` function in the js file.
not sure if this is the most usable solution, but at least is solves my problem.
https://github.com/CMB2/CMB2/compare/trunk...bomsn:patch-1

The solution was based on another fork by @nonsensecreativity https://github.com/nonsensecreativity/CMB2/commit/a15bf588171d5f26de71af088b6f594da9f84679

There is another issue tough, when trying to remove a field from empty group, it will create new empty field in previous group.

hope you can tackle these in the next release.

**Comment by tw2113** [comment]
If I recall right, CMB2 will visually provide at least 1 field, though it won't, or at least shouldn't, save anything with it visually shown. Only once items start getting filled in would anything save.

**Comment by bomsn** [comment]
@tw2113 That is correct, I probably didn't explain this clearly. The issue I had is that the JS was not giving IDs/Names correctly which was causing issues when saving the post, similar issues was posted before:
https://github.com/CMB2/CMB2/issues/116
https://github.com/CMB2/CMB2/issues/521

I solved this, but I found another issue with `cmb.removeAjaxRow` function, when we have 3 groups for example, say first group have 3 field and second group have 2, and the 3rd is still empty (will have an empty field), say you clicked 'remove button' on the empty field inside the third group, this will trigger `cmb.resetRow`, and this function will trigger a click on remove row button and add row button at the same time, remove row trigger seem to be working correctly, however, add row button is not triggered within the third group only, but all other group, which will create additional field in all groups.

**Comment by tw2113** [comment]
@jtsternberg You have any ideas here?

**Comment by Mr2P** [comment]
Thank @bomsn, I got the same issue. Your patch worked for me. 
Anyway,  the group index is correct but the field index is still not correct. In my case, when creating a new group (the fourth group), the ID of the first row of the repeatable field is: `group_name[3][field_name][0][nested_field_name], the next row is group_name[3][field_name][1][nested_field_name]` but the third row gets the ID is `group_name[3][field_name][4][nested_field_name]`

**Comment by bomsn** [comment]
@Mr2P Make sure to use the patch on the trunk version.  If it still doesn't work please share the code for rendering you custom field. Also does it work correctly on the 3 first groups, or you get the same issue on all groups?

**Comment by Mr2P** [comment]
@bomsn Your patch solved my issue. I only needed the group index is correct for my hotfix.

**Comment by bomsn** [comment]
@Mr2P Happy it helped.
